mutt/string.c

```diff
@@ -959,3 +959,71 @@ const char *mutt_str_getenv(const char *name)
 
   return NULL;
 }
+
+/**
+ * Remove all occurrences of substring, ignoring case.
+ * @param str     String containing the substring
+ * @param target  Target substring for removal
+ * @retval 0      String contained substring and substring was removed successfully.
+ * @retval != 0   String did not contain substring.
+ */
+int mutt_str_remall_strcasestr(char *str, const char *target)
+{
+  // There's no point in removing something that is not there.
+  if (mutt_str_strcasestr(str, target) == NULL)
+    return 1;
+
+  // Look through an ensure all instances of the substring are gone.
+  while ((str = (char *) mutt_str_strcasestr(str, target)))
+  {
+    size_t target_len = mutt_str_strlen(target);
+    memmove(str, str + target_len, 1 + strlen(str + target_len));
```

Comment:
... and set a flag here, to determine the return value

Reply:
I am unsure how to achieve this due to my lack of familiarity with C. Could you provide an example or link to an example?

Reply:
Okay, I think I might have figured out how to achieve this.

Would it be something like:

```C
int retval = 0;

while ((retval = (str = (char *) mutt_str_strcasestr(str, target))))
{
   // loop body
}

return retval;
```

That seems mess-y and hard to read, but compiles and works; however, I believe it is abuse of the standard (and some poor, future individual's sanity.)